framework/src/outputs/XDA.C

```diff
@@ -44,15 +44,17 @@ XDA::XDA(const InputParameters & parameters) :
 void
 XDA::output(const ExecFlagType & /*type*/)
 {
+  std::string fname = filename();
+  std::string base = fname.substr(0, fname.size()-4);
   if (_binary)
   {
-    _mesh_ptr->getMesh().write(filename()+"_mesh.xdr");
-    _es_ptr->write (filename()+".xdr", ENCODE, EquationSystems::WRITE_DATA | EquationSystems::WRITE_ADDITIONAL_DATA);
+    _mesh_ptr->getMesh().write(base + "_mesh.xdr");
```

Comment:
@permcody I fixed this again to use a std::string insert, it is more better.

On Wed, Mar 9, 2016 at 4:13 PM, Cody Permann <notifications@github.com>
wrote:

> In framework/src/outputs/XDA.C
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_idaholab_moose_pull_6544-23discussion-5Fr55607746&d=BQMCaQ&c=54IZrppPQZKX9mLzcGdPfFD1hxrcB__aEkJFOKJFd00&r=h7heP8xwI1i_HikChvhFbEBurKirgfOCdwgBxB9lM8c&m=AMOCY7FGymNuxJA4135T3g7qHDpMROL2nljxhWXspU4&s=4V06JhM0r-vxonmhZMH7YHr3GV2cJdGx_1HnElPikOo&e=>
> :
>
> >    if (_binary)
> >    {
> > -    _mesh_ptr->getMesh().write(filename()+"_mesh.xdr");
> > -    _es_ptr->write (filename()+".xdr", ENCODE, EquationSystems::WRITE_DATA | EquationSystems::WRITE_ADDITIONAL_DATA);
> > +    _mesh_ptr->getMesh().write(base + "_mesh.xdr");
>
> I'm just saying that you have duplicated logic in both methods
>
> if (_binary)
>     // Append ".xdr";
>   else
>     // Append ".xda";
>
> I can merge as is, it just interesting that you throw the suffix away and
> then have it hardcoded to put it back on again.
>
> —
> Reply to this email directly or view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_idaholab_moose_pull_6544_files-23r55607746&d=BQMCaQ&c=54IZrppPQZKX9mLzcGdPfFD1hxrcB__aEkJFOKJFd00&r=h7heP8xwI1i_HikChvhFbEBurKirgfOCdwgBxB9lM8c&m=AMOCY7FGymNuxJA4135T3g7qHDpMROL2nljxhWXspU4&s=QmqMSqp3mwUw2Ga3JXGpIGsltalJw7mOUWxAgCM6y10&e=>
> .
>